### ISSUE TITLE
yubioath-desktop not available in Ubuntu 19.10 eoan

### ISSUE THREAD

**Issue by KlfJoat** [Yubico/yubioath-desktop#479]
- Yubico Authenticator version: N/A
- Operating system and version: 
```
$ uname -a
Linux hostname 5.3.0-18-generic #19-Ubuntu SMP Tue Oct 8 20:14:06 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 19.10
Release:        19.10
Codename:       eoan
```
- YubiKey model and version: N/A
- Bug description summary: yubioath-desktop package not available in Ubuntu 19.10, either in Universe repository or in the yubico/stable PPA

Steps to reproduce
---
```
$ apt search yubioath
Sorting... Done
Full Text Search... Done
$
```
The following URLs show no sign of yubioath-desktop
* https://launchpad.net/~yubico/+archive/ubuntu/stable?field.series_filter=eoan
* https://packages.ubuntu.com/eoan/yubioath-desktop
* https://packages.ubuntu.com/source/eoan/yubioath-desktop

Expected result
---
yubioath-desktop being available for my use in Ubuntu.

Actual results
---
```
$ sudo apt install yubioath-desktop
Reading package lists... Done
Building dependency tree
Reading state information... Done
E: Unable to locate package yubioath-desktop
$
```

Other info
---
Google results show info from the packages.ubuntu.com links above, so there was information there at one time???

**Comment by KlfJoat** [comment]
Might I suggest that the PPA description be updated to reflect this change?  Because I assumed that I would be able to continue getting it as I had been and I read nothing to disabuse me of that notion.

Also, what about updates of this AppImage package?  One of the things I adore about Linux over Win or Mac is automatic updates and upgrades.  Snaps and Flatpaks auto-update, or auto-updates can be run.

**Comment by KlfJoat** [comment]
Also, how can I get an AppImage executable on my Gnome-shell dock?  Or in the Gnome Applications list?  

This whole AppImage thing may make things easier for you, but it is a definite step back in functionality for me.

**Comment by dc602003** [comment]
I too would like to request that this be addressed. I have downloaded and executed the Yubikey-Manager, but I used to use the yubioath-desktop to view and copy out the OTP codes on my laptop. Could this functionality be added or built into a similar app, please?

**Comment by braathen** [comment]
Thanks guys, we understand your pain but I'm afraid I have to refer to what Dag is saying above. The AppImage is currently the only distribution format we are able to provide at this time, but if the community is willing to do an effort in packaging we will try to help out in any way we can.

@KlfJoat As an alternative maybe enabling the app to show in the system tray could work for you? This will also enable favorited accounts to show up there, making it easier to quickly access them.

@dc602003 did you try the [AppImage](https://developers.yubico.com/yubioath-desktop/Releases/)? Yubico Authenticator is very much still available and in active development, it's just not available from a PPA repository anymore.

**Comment by sgabay242** [comment]
Hello, this has been a problem for me as well, installing via apt on Ubuntu 18 has been no issue but when moving to ubuntu 19 authenticator is not available in the repo, per the documentation it is expected for the entire set of apps to be available but after installing it is not. Building from source is not straightforward, the QT stack requires LOTS of dependencies, I could not make this work, In my opinion it is unreasonable and a security issue to expect to have an entire development environment to compile a the Yubi Authenticator on linux laptops. lets put it this way...it does not scale and does not help Yubico adoption on linux, unfortunately this is definitely a step back.Further it is inconsistent to have Yubikey-manage and the personalization tool but not the authenicator to complete the set, even the entire set is available for ubuntu 18. This would be a big win to be available via apt for the future due to the streamlined installation process. Thank you

**Comment by dc602003** [comment]
@braathen Thanks you for responding. Yes, I tried the AppImage, however this is good for managing the YubiKey, I could not find anywhere in the app to load the OTPs from the inserted YubiKey. Are you sure it's contained in the App?

**Comment by braathen** [comment]
@dc602003 it sounds like you are using YubiKey Manager and not Yubico Authenticator? Try the link I referred to above.

OK... no promises, but after some internal discussions we said that we will at least look at what we can do about the distribution. Thanks for hanging in there in the meantime and it IS encouraging to hear from you all! :)

**Comment by dc602003** [comment]
Thanks @braathen , I used your link, and that AppImage is indeed the Yubico Authenticator. Not sure how I managed to get the Manager by mistake.  Thank you for looking into an alternative method of distribution too, it's appreciated.

**Comment by rasor** [comment]
A little guide for Linux dummies like myself:  
Avoiding error `E: Unable to locate package yubioath-desktop`:

* Open Software Sources
* Goto tab Other Software - you should have a line with http://ppa.launchpad.net/yubico/stable/ubuntu eoan main.  
But yubico on [eoan (19.10) main does not include Yubico Authenticator](https://launchpad.net/%7Eyubico/+archive/ubuntu/stable/+index?field.series_filter=eoan)  
So we [need an older one from disco (19.04)](https://launchpad.net/~yubico/+archive/ubuntu/stable/+index?field.series_filter=disco)
* In Other Software - de-select the eoan version  
* Add `deb http://ppa.launchpad.net/yubico/stable/ubuntu disco main` and select it

```bash
# update sw from Other Software (and all the other sources)
sudo apt update
# you should now be able to find the authenticator in your repos
apt list yubi*
# if you got it in the list you can install it
sudo apt install yubioath-desktop
apt list yubi*
# yubioath-desktop/disco,now 4.3.6~ppa1~disco1 amd64 [installed]
```
Thanks to @braathen

**Comment by Maryse47** [comment]
There is attempt (unofficial for now) to create flatpak package: https://github.com/flathub/flathub/pull/1126

**Comment by iam-TJ** [comment]
What's the (internal) resistance to re-adding yubioath-desktop to the 19.10+ PPA (or, why was it removed) ? As we're headed for the 20.04 LTS release very soon it would be good to get this sorted in time for that.

As the PPA seems to contain most (all?) of the other packages as were in pre-19.10 releases presumably some issue caused yubioath-desktop to be dropped?

Maybe something was lost to do with debian-specific packaging, or build tooling to support that?

I've been a long-time Ubuntu dev across multiple packages and would be prepared to work on getting the debian packaging back into line so it could be pushed to the PPA for building by the PPA team.

**Comment by dagheyman** [comment]
The main motivation for us to move away from the PPA is that it allows us decide what version of Qt we want to depend on, and not be held back depending on what is available in the oldest Ubuntu LTS we want to support. For non GUI projects this is not as much of an issue.

We probably won't be re-adding this app to our PPA, but we are looking in to Snap packages which would allow us to use our own Qt, and third parties are working on [native debian packaging](https://github.com/Yubico/yubioath-desktop/issues/328) and [Flatpak](https://github.com/flathub/flathub/pull/1126). Maybe you could help out with the debian package, which would then eventually be included in the Ubuntu repositories?

**Comment by Maryse47** [comment]
@dagheyman if the qt version is such important then why you didn't chose flatpak which provides any qt from 5.9 to 5.14 in corresponding runtime rather than having to bundle your own qt in snap? Do you need some downstream patches in qt?

**Comment by berlinguyinca** [comment]
any chance to get this fixed?

**Comment by taowa** [comment]
I've asked to have the version in sid synced to Ubuntu. The request was processed and it's currently in the Ubuntu NEW queue for Focal (20.04).

**Comment by taowa** [comment]
@dagheyman  see https://launchpad.net/ubuntu/+source/yubioath-desktop.

As far as I can tell (though I'm less comfortable with Ubuntu, so I could be wrong), there's no way to get it into 19.10, but it'll be in the 20.04 LTS.

Taowa

**Comment by iam-TJ** [comment]
Regarding Ubuntu 19.10: there's not much point backporting the package from 20.04 since the non-LTS release support window is only 9 months [0] so 19.10 will be end-of-life in July.

If being tied to the GUI framework is so important then the obvious solution is to separate the core functionality into a user 'service' (local TCP socket ?) and a GUI client. This would also allow other client frontends including HTTP/HTML.

I'd be up for working on such a solution over the coming months.

[0] https://ubuntu.com/about/release-cycle

**Comment by taowa** [comment]
@iam-TJ In my understanding, someone correct me if I'm wrong, that's basically what ykman provides. yubioath-desktop is effectively just the GUI for it since `ykman oath` will do the same things on the command line and yubioath-desktop is [just making calls to ykman anyway.](https://github.com/Yubico/yubioath-desktop/blob/master/py/yubikey.py#L15)

While I'm sure an HTTP or HTML framework would not be unwelcome, I also think it's far beyond the scope of this issue, as far as I can tell.

**Comment by dagheyman** [comment]
Closing this, looks like the package will be available in Ubuntu 20.04 (thanks @taowa !)
